maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java

```diff
@@ -526,6 +563,45 @@ private static StringBuilder addHeaderAttribute(StringBuilder builder, String ke
 		return builder;
 	}
 
+	private void attachArtifactToProject(Jar bndJar) {
+		File artifactFile = createArtifactFile();
+
+		try (OutputStream os = buildContext.newFileOutputStream(artifactFile)) {
+			bndJar.write(os);
+
+			Optional<String> classifier = getClassifier();
+
+			// If there is a classifier artifact must be attached to the
+			// project using that classifier
+			if (classifier.isPresent()) {
+				projectHelper.attachArtifact(project, getType().orElse(""), classifier.get(), artifactFile);
```

Comment:
At this point, the outputstream is not closed and so the artifactFile is still open. Perhaps the outputstream should be closed before artifactFile is used? That is, tighten up the try-with-resources block so this code is after it.